content/client-server-api/modules/end_to_end_encryption.md

```diff
@@ -1596,6 +1624,13 @@ that they can decrypt future messages encrypted using this session. An
 `m.room_key` events sent by other devices in order to decrypt their
 messages.
 
+When a client is updating a Megolm session (room key) in its store, the client MUST ensure:
+
+* that the updated session data comes from a trusted source, where a trusted source
+  could be the signatures on the session or those from the user's session key backup;
+  and
```

Comment:
I think this is not quite right. The source should have an equal or higher trust, but the online key backup for example is not a trusted source, since anyone can upload sessions to it.

Reply:
I mean, this is straight from the MSC almost to the letter. It's possible it's considering it trusted because if the key works to decrypt the message then it's clearly valid enough, however that doesn't necessarily solve the concern.

Also, I don't think anyone except the user holding the keys to the backup can write to it? Smells like a security issue that shouldn't be discussed here if people can arbitrarily write to key backups.

Reply:
It's publicly known; see the intro to the [Symmetric Megolm Backup MSC](https://github.com/uhoreg/matrix-doc/blob/symmetric-backups/proposals/3270-symmetric-megolm-backup.md#msc3270-symmetric-megolm-backup).

I'm guessing Erik just wasn't aware of this when drafting the MSC and no one caught it. In the future we'll have symmetric backups which shouldn't have this problem.

Reply:
I suppose that means we drop the key backup from here? Not really sure what the conclusion is...

Reply:
Perhaps let's just say the following, without giving an example for now:

> that the updated session data comes from a trusted source

Reply:
> Perhaps let's just say the following, without giving an example for now:
>> that the updated session data comes from a trusted source

This sounds reasonable to me.